skycfg.go

```diff
@@ -426,11 +426,26 @@ func (c *Config) Main(ctx context.Context, opts ...ExecOption) ([]proto.Message,
 	var msgs []proto.Message
 	for ii := 0; ii < mainList.Len(); ii++ {
 		maybeMsg := mainList.Index(ii)
-		msg, ok := AsProtoMessage(maybeMsg)
-		if !ok {
-			return nil, fmt.Errorf("%q returned something that's not a protobuf (a %s)", parsedOpts.funcName, maybeMsg.Type())
+		// Only flatten but not flatten deep. This will flatten out, in order, lists within main list and append the
```

Comment:
To avoid breaking any other existing uses of skycfg... would you be against making an ExecOption to turn on list flattening? I imagine just add a `flattenLists bool` to the execOptions struct, and then a:
```
// WithFlattenLists flattens lists one layer deep ([1, [2,3]] -> [1, 2, 3])
func WithFlattenLists() ExecOption {
	return fnExecOption(func(opts *execOptions) {
		opts.flattenLists = true
	})
}
```

Then you can case the logic below on `parsedOpts.flattenLists`

Reply:
updated